src/rbus/rbus.c

```diff
@@ -2262,6 +2262,15 @@ static void _subscribe_callback_handler (rbusHandle_t handle, rbusMessage reques
                 RBUSLOG_ERROR("%s: payload missing in subscribe request for event %s from %s", __FUNCTION__, event_name, sender);
             }
 
+            elementNode* el = NULL;
+            el = retrieveInstanceElement(handleInfo->elementRoot, event_name);
+
+            if (el !=NULL && (el->type == RBUS_ELEMENT_TYPE_TABLE)  && (event_name[strlen(event_name)-1] != '.'))
+            {
+                RBUSLOG_ERROR(":%s: Invalid event_name: %s, Element Table Subscription should end with '.'",__FUNCTION__, event_name);
+                return ;
```

Comment:
1. Do not return here and instead set the return code as failure. So that the consumer wont add it to its subscription list.
2. Use the `el`  passed as input to `_event_subscribe_callback_handler()` (first argument) and use it instead of doing `retrieveInstanceElement()` once again there.
3. Also use the same `el`  when `publishOnSubscribe` case and avoid another retrieval.

Reply:
Addressed all review comment points.